cmd/swapd/main_test.go

```diff
@@ -137,6 +137,12 @@ func TestDaemon_PersistOffers(t *testing.T) {
 	startupTimeout := time.Millisecond * 100
 
 	dataDir := t.TempDir()
+	defer func() {
+		// CI has issues with the filesystem still being written to when it is
+		// recursively deleting dataDir. Can't be replicated outside of CI.
+		unix.Sync()
+		time.Sleep(500 * time.Millisecond)
+	}()
```

Comment:
Line 192 below calls `d.stop()`, so I don't know what could be writing to the filesystem when the `dataDir` is recursively deleted:
```
testing.go:1097: TempDir RemoveAll cleanup: unlinkat /tmp/TestDaemon_PersistOffers1622151463/001: directory not empty
```
Given that I can't replicate it outside of CI, my theory is that the CI filesystem may just be crappy and still flushing stuff to disk. I tried running it a very large number of times outside of CI, so I could capture what was in the directory, but couldn't get a failure.